src/VirtoCommerce.AssetsModule.Core/Provider/BasicBlobProvider.cs

```diff
@@ -16,11 +16,29 @@ protected BasicBlobProvider(IOptions<PlatformOptions> platformOptions, ISettings
             _settingsManager = settingsManager;
         }
 
+        // Blacklisted or whitelisted extension
         public virtual bool IsExtensionBlacklisted(string path)
         {
             var blackList = _platformOptions.Value.FileExtensionsBlackList.Union(
                 _settingsManager?.GetObjectSettingAsync(PlatformConstants.Settings.Security.FileExtensionsBlackList.Name).Result.AllowedValues.Cast<string>() ?? new string[0]);
-            return (blackList.Any(x => path.Trim().EndsWith(x.Trim(), StringComparison.OrdinalIgnoreCase)));
+            var blacklisted = (blackList.Any(x => path.Trim().EndsWith(x.Trim(), StringComparison.OrdinalIgnoreCase)));
+            var blackListCount = blackList.Count();
+
+            var whiteList = _platformOptions.Value.FileExtensionsWhiteList.Union(
+                _settingsManager?.GetObjectSettingAsync(PlatformConstants.Settings.Security.FileExtensionsWhiteList.Name).Result.AllowedValues.Cast<string>() ?? new string[0]);
+            var whitelisted = (whiteList.Any(x => path.Trim().EndsWith(x.Trim(), StringComparison.OrdinalIgnoreCase)));
+            var whiteListCount = whiteList.Count();
+
+            var result = false;
+            if(blacklisted) result = true;
+            else if(blackListCount == 0)
+            {
+                if (whiteListCount == 0) result = false;
+                else if (whitelisted) result = false;
+                else result = true;
+            }
+
+            return result;

```

Comment:
Delete `result`. Optimize `ifs`.
```suggestion
            if (blacklisted) return true;
            if (blackListCount != 0 || whiteListCount == 0) return false;
            
            return !whitelisted;
```